paper_framework_sources/p1897_initial_executor_algorithms_for_cpp23.md

```diff
@@ -694,21 +748,123 @@ float r = sync_wait(
+   When some `output_receiver` has been passed to `connect` on the returned `sender` returning some `operation_state` `os2`:
 
- * Otherwise constructs a receiver, `r` over an implementation-defined synchronization primitive and passes that receiver to `execution::submit(S, r)`.
+   * If `set_value(r, ts...)` is called, passes `ts...` to `set_value(output_receiver, ts...)`.
+   * If `set_error(r, e)` is called, calls `std::invoke(f, ts...)` to return some `invoke_result`, and calls `execution::start(execution::connect(invoke_result, output_receiver))`.
+   * If `f` throws, catches the exception and passes it to `set_error(output_receiver, e)`.
+   * If `set_done(r)` is called, calls `set_done(output_receiver)`.
+
+   When `start` is called on `os2`, call `execution::start(os)`.
+
+ * Otherwise the expression `execution::handle_error(s, f)` is ill-formed.
+
+
+## execution::share
+
+### Overview
+`share` is a sender adapter that takes a `sender`, eagerly submits it and returns a `sender` that propagates the value by reference and can be used as an l-value.
+
+Signature:
+```cpp
+template <execution::sender S>
+see-below share(S s);
+```
+
+*[ Example:*
+```cpp
+auto s1 = just(3) | share();
+auto s2 = s1 | transform([](const int& v){return v+1;}))
+auto s3 = s1 | transform([](const int& v){return v+2;}))
+int r = sync_wait(
+  transform(
+    when_all(s2, s3),
+    [](int a, int b){return a+b;}));
+// r==9
+```
+
+### Wording
+The name `execution::share` denotes a customization point object.
+For some subexpressions `s`, let `S` be a type such that `decltype((s))` is `S`.
+The expression `execution::share(s, f)` is expression-equivalent to:
+
+ * `s.share()` if that expression is valid and if `s` satisfies `sender`.
+ * Otherwise, `share(s)`, if that expression is valid, if `s` satisfies `sender` with overload resolution performed in a context that includes the declaration
+ ```
+    void share() = delete;
+ ```
+   and that does not include a declaration of `execution::share`.
+
+ * Otherwise constructs a receiver, `r` and passes that receiver to `execution::connect(s, r)` resulting in an `operation_state` `os`.
+   Constructs some shared state, `shr` to store the completion result(s) of `s`.
+
+   * If `set_value(r, ts)` is called stores `ts` in `shr`.
+   * If `set_error(r, e)` is called, stores `e` in `shr`.
+   * If `set_done(r)` is called stores the done result in `shr`.
+
+   When some `output_receiver` has been passed to `connect` on the returned `sender`, resulting in an `operation_state` `os2` and one of the above has been called on `r`:
+   * If `r` was satisfied with a call to `set_value`, call `set_value(output_receiver, ts...)`
+   * If `r` was satisfied with a call to `set_error`, call `set_error(output_receiver, e)`.
+   * If `r` was satisfied with a call to `set_done`, call `execution::set_done(output_receiver)`.
+
+ * When `start` is called on `os2`, call `execution::start(os)`.
+
+## execution::let
```

Comment:
based on this description, I would say that let is transform_value()

Reply:
Naming.

Is transform not transform_value?

Reply:
the difference is that transform does not connect the return value. transform_value does connect tot he return value which allows transform_value to switch to a different signal, not just a different value.

Reply:
No, I understand the algorithm difference. I think the similarity of the names is confusing if transform has very different behaviour, it isn't clear why the _value suffix implies that.

Should we rename let to transform_value in unifex to be consistent with transform_done?

I can do the rename, though, I agree consistency is good.

Reply:
let_value, let_error (and potentially let_done in the future) agreed on for consistency here.

Reply:
I did the rename in the latest push.